ext/zip/config.m4

```diff
@@ -4,36 +4,21 @@ PHP_ARG_WITH([zip],
     [Include Zip read/write support])])
 
 if test "$PHP_ZIP" != "no"; then
-  PKG_CHECK_MODULES([LIBZIP], [libzip >= 0.11])
-  LIBZIP_LIBDIR=`$PKG_CONFIG --variable=libdir libzip`
+  PKG_CHECK_MODULES([LIBZIP], [libzip >= 1.0.1])
```

Comment:
Why ?

Reply:
@remicollet, as I noted above, we currently check for the `zip_open` function:
`PHP_CHECK_LIBRARY(zip, zip_open, ...`
If this function is not found, `configure` aborts.

Now, `zip_open` was added in libzip 1.0, so I don't see the point in allowing older versions, since we abort anyway.

Reply:
No, zip_open is there since ever.

Minimal version is 0.11 which is the first one to provide the needed public API which was private in old 0.10

Reply:
@remicollet, I checked the 0.11 release and `zip_open` is definitely there. It's somewhat strange, then, that https://libzip.org/documentation/zip_open.html says libzip 1.0.

That said, is there a need to maintain support for the minimum needed API as 0.11? Can we raise the minimum required?

Reply:
1/ I don't see any benefit do drop this support

2/ I don't like mixing different stuff totally unrelated to each other in a single commit

Reply:
I suggest to stick with the minimum requirement, but to drop the zip_open() check nonetheless, because it is pointless.

Reply:
@remicollet, any objections to @cmb69's suggestion to drop the `zip_open` check? Also, I'm not sure what you mean by unrelated. There are four commits (although that will probably change following feedback from these reviews).

Reply:
The CHECK_LIBRARY on some old (always there) symbols is there to ensure usability of the library., like 
  PHP_CHECK_LIBRARY(ffi, ffi_call,
  PHP_CHECK_LIBRARY(bz2, BZ2_bzerror,
  PHP_CHECK_LIBRARY(aspell,new_aspell_config,
...

> I'm not sure what you mean by unrelated

I mean if the PR plan to fix a bug, it should only fix the bug, and no unrelated change.

Cleanup commit can be part of another PR.

Reply:
@nikic, any thoughts on removing the `zip_open` check?

Reply:
Per current convention the zip_open check should be dropped -- pkg-config output is trusted. Previously it was needed because we were doing guesswork on the library location, and guesswork can be wrong ;)